component/file_cache/file_cache_test.go

```diff
@@ -874,6 +874,60 @@ func (suite *fileCacheTestSuite) TestCloseFileTimeout() {
 	suite.assert.True(err == nil || os.IsExist(err))
 }
 
+func (suite *fileCacheTestSuite) TestOpenCloseHandleCount() {
+	defer suite.cleanupTest()
+	// Setup
+	file := "file14"
+	handle, err := suite.fileCache.CreateFile(internal.CreateFileOptions{Name: file, Mode: 0777})
+	suite.assert.NoError(err)
+	err = suite.fileCache.CloseFile(internal.CloseFileOptions{Handle: handle})
+	suite.assert.NoError(err)
+
+	handle, err = suite.fileCache.OpenFile(internal.OpenFileOptions{Name: file, Mode: 0777})
+	suite.assert.NoError(err)
+	err = suite.fileCache.CloseFile(internal.CloseFileOptions{Handle: handle})
+	suite.assert.NoError(err)
+
+	// check that flock handle count is correct
+	flock := suite.fileCache.fileLocks.Get(file)
+	suite.assert.Zero(flock.Count())
+}
+
+func (suite *fileCacheTestSuite) TestOpenPreventsEviction() {
+	defer suite.cleanupTest()
+	// Setup
+	suite.cleanupTest() // teardown the default file cache generated
+	cacheTimeout := 1
+	config := fmt.Sprintf("file_cache:\n  path: %s\n  offload-io: true\n  timeout-sec: %d\n\nloopbackfs:\n  path: %s",
+		suite.cache_path, cacheTimeout, suite.fake_storage_path)
+	suite.setupTestHelper(config) // setup a new file cache with a custom config (teardown will occur after the test as usual)
+
+	path := "file14"
```

Comment:
Should this be a different filename, to follow the other tests?

Reply:
Oh yeah!